### ISSUE TITLE
0/0 -> 100% coverage with php 7.0 - xdebug Xdebug 2.4.0RC4

### ISSUE THREAD

**Issue by raziel057** [sebastianbergmann/php-code-coverage#422]
When I run my tests with PHP 7.0.0 with phpunit 5.2.0 and xdebug 2.4.0RC4 I got this bad result:

![code-coverage](https://cloud.githubusercontent.com/assets/652505/13072036/6f78a588-d497-11e5-8692-e5b0797f5b2a.png)

When I run with php 5.6.11, with the same version of phpunit and xdebug 2.3.3, I got this expected result:

![code-coverage-php5 6](https://cloud.githubusercontent.com/assets/652505/13108629/934979d2-d574-11e5-806e-10e9c6a63bfd.png)

**Comment by sebastianbergmann** [comment]
Before I investigate this further I would like to make sure that this only depends on the version of PHP. Can you please at least keep the versions of PHPUnit and PHP_CodeCoverage the same? In the one screenshot you currently have PHP 7.0 / PHPUnit 5.2.5 / PHP_CodeCoverage 3.2.0 and in the other PHP 5.6 / PHPUnit 5.1.3 / PHP_CodeCoverage 3.0.2.

**Comment by raziel057** [comment]
I've just run the tests with the exact same version of phpunit to check if we have the same behavior. I updated my previous issue description.

**Comment by KevinMace** [comment]
I am also having the same issue as @raziel057 , these are the versions of PHPUnit and PHP_CodeCoverage i'm using when having the issue.
```
Generated by PHP_CodeCoverage 2.2.4 using PHP 7.0.1 and PHPUnit 4.8.23
```

**Comment by Rockstar04** [comment]
Just want to pop in with the same issue. Spent a couple hours compiling different versions of PHP, xDebug and trying to figure out if it was an issue with my config files.

![PHP Code Coverage Discrepancy](http://i.imgur.com/pNiCVZ7.png)

**Comment by sebastianbergmann** [comment]
I did some investigation last week and it looks to me like an issue with Xdebug and PHP 7.

**Comment by sebastianbergmann** [comment]
Would be great if somebody could check whether the problem persists with the current `master` of Xdebug as @derickr fixed a few code coverage related issues since Xdebug 2.4.0RC4.

**Comment by Rockstar04** [comment]
Building Xdebug from the current master did not change the coverage results.

```
PHP 7.0.3 (cli) (built: Feb 29 2016 06:34:53) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Xdebug v2.4.0RC5-dev, Copyright (c) 2002-2016, by Derick Rethans
```

**Comment by Rockstar04** [comment]
Tested again with the following:

```
PHP 7.0.3 (cli) (built: Feb 29 2016 06:34:53) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Xdebug v2.4.1-dev, Copyright (c) 2002-2016, by Derick Rethans
```

```
PHP 7.0.4 (cli) (built: Mar  8 2016 08:37:55) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Xdebug v2.4.1-dev, Copyright (c) 2002-2016, by Derick Rethans
```

Coverage results are still the same as above with RC4

**Comment by lordelph** [comment]
Just to add I'm seeing this behaviour too
* PHP_CodeCoverage 3.3.0
* PHP 7.0.4-5+deb.sury.org~wily+1
* PHPUnit 5.2.10 
* XDebug 2.4.0-1+deb.sury.org~wily+1

**Comment by skors** [comment]
Same behavior here with
* php-code-coverage 4.0.0 
* PHP 7.0.5-3+donate.sury.org~trusty+1 
* Xdebug 2.4.0 
* PHPUnit 5.4.6

**Comment by GrahamCampbell** [comment]
Try with PHP 7.0.8. Earlier versions of PHP have known bugs with code coverage.

**Comment by Rockstar04** [comment]
I still have large discrepancies between 7.0.8 and 5.6.19

**Comment by skors** [comment]
updated to php 7.0.8 but shows same problem

are there any further informations interesting to give?
I am also experiencing problems with ```xdebug_start_code_coverage(XDEBUG_CC_DEAD_CODE | XDEBUG_CC_UNUSED);``` which leads to SIGSEGV - core dumped errors from FPM process, i only can use one of the two constant options (but this seems to be a xdebug issue)

**Comment by jaccoTopVintage** [comment]
Still got this issue with PHP 7.0.8 (VC14 x64 Thread Safe (2016-Jun-22 19:17:02))

**Comment by GrahamCampbell** [comment]
Are you able to install the latest dev version of xdebug?

**Comment by skors** [comment]
@GrahamCampbell thank you very much, this fixes the Xdebug constants options problem when starting coverage

The issue of this thread also exists with 
* PHP 7.0.8
* Xdebug 2.5.0-dev
* PHPUnit 5.4.6
* PHP_CodeCoverage 4.0.0

**Comment by micgro42** [comment]
This bugreport for xdebug seems top be very related: [phpunit coverage with php7 and 2.4.0RC3-dev coverage always 100%](https://bugs.xdebug.org/view.php?id=1224)
It seems that the problem is actually **not** in xdebug?

It also mentions the workaround to use [``@runInSeparateProcess``](https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.runInSeparateProcess). Maybe that helps some people?

**Comment by MatthiasKuehneEllerhold** [comment]
I've ran into the same problem with PHPUnit 5.4.6 , PHP 7.0.8-1~dotdeb+8.1 and Xdebug 2.4.0.
If I'm using PHP-DBG instead of Xdebug it works, but my unit tests are 50% slower.

**Comment by samsonasik** [comment]
duplicate with #411

**Comment by voycey** [comment]
I'm also having this problem since upgrading to PHP7:

PHP 7.0.10-2+deb.sury.org~xenial+1
Xdebug v2.4.1
Code Coverage 4.0.1
PHP Unit 5.5.2

![image](https://cloud.githubusercontent.com/assets/1065098/18540968/83c3e780-7b65-11e6-904f-5e3a25a51443.png)
![image](https://cloud.githubusercontent.com/assets/1065098/18540972/8a40153e-7b65-11e6-9da0-8795ad0542e4.png)

**Comment by romainneutron** [comment]
Should be closed in favor of #411

**Comment by JoepRoebroek** [comment]
@raziel057 I proposed a fix for this issue in #488. Could you check if this fix also solves the problem you are having in this issue?

**Comment by raziel057** [comment]
@JoepRoebroek Thanks! I will try to check it next week.

**Comment by lordelph** [comment]
The fix proposed in #488 will address this. You will need to build the current master branch of xdebug, or wait until a new binary release is made.
